lib/pdk/util/bundler.rb

```diff
@@ -93,42 +115,45 @@ def installed?(gem_overrides = {})
           result[:exit_code].zero?
         end
 
-        def lock!(gem_overrides = {})
+        def lock!
           if PDK::Util.package_install?
             # In packaged installs, use vendored Gemfile.lock as a starting point.
             # Subsequent 'bundle install' will still pick up any new dependencies.
-            vendored_gemfile_lock = File.join(PDK::Util.package_cachedir, 'Gemfile.lock')
+            vendored_lockfiles = [
+              File.join(PDK::Util.package_cachedir, "Gemfile-#{PDK::Util::RubyVersion.active_ruby_version}.lock"),
+              File.join(PDK::Util.package_cachedir, 'Gemfile.lock'),
+            ]
+
+            vendored_gemfile_lock = vendored_lockfiles.find { |lockfile| File.exist?(lockfile) }
 
-            unless File.exist?(vendored_gemfile_lock)
+            unless vendored_gemfile_lock
               raise PDK::CLI::FatalError, _('Vendored Gemfile.lock (%{source}) not found.') % {
                 source: vendored_gemfile_lock,
               }
             end
 
             PDK.logger.debug(_('Using vendored Gemfile.lock from %{source}.') % { source: vendored_gemfile_lock })
             FileUtils.cp(vendored_gemfile_lock, File.join(PDK::Util.module_root, 'Gemfile.lock'))
-
-            # Update the vendored lock with any overrides
-            update_lock!(gem_overrides, local: true) unless gem_overrides.empty?
           else
             argv = ['lock']
 
             cmd = bundle_command(*argv).tap do |c|
-              c.add_spinner(_('Resolving Gemfile dependencies.'))
-              c.update_environment(gemfile_env(gem_overrides)) unless gem_overrides.empty?
+              c.add_spinner(_('Resolving default Gemfile dependencies.'))
+              # c.update_environment(gemfile_env(gem_overrides)) unless gem_overrides.empty?
```

Comment:
This commented needs to be removed or a reason for leaving it should be documented.

Reply:
Oh yeah I've been meaning to remove this line.